tests-e2e/core/managed_environment_status_test.go

```diff
@@ -0,0 +1,233 @@
+			Expect(condition.Reason).To(Equal("UnableToParseKubeconfigData"))
+			Expect(condition.Message).To(ContainSubstring("json parse error"))
+		})
+
+		It("should have a connection status condition of False when there is no current contex in the kubeconfig", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("creating the GitOpsDeploymentManagedEnvironment with a secret that lacks a kubeconfig context")
+
+			kubeConfigContents, apiServerURL, err := extractKubeConfigValues()
+			Expect(err).To(BeNil())
+			managedEnv, secret := buildManagedEnvironment(apiServerURL, kubeConfigContents)
+			secret.StringData["kubeconfig"] = "apiVersion: v1\nkind: Config\n"
+
+			k8sClient, err := fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			err = k8s.Create(&secret, k8sClient)
+			Expect(err).To(BeNil())
+
+			err = k8s.Create(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+
+			By("ensuring the managed environment connection status condition has a reason of UnableToLocateContext")
+			Eventually(managedEnv, "2m", "1s").Should(HaveStatusCondition(managedgitopsv1alpha1.ManagedEnvironmentStatusConnectionInitializationSucceeded))
+			err = k8s.Get(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+			Expect(managedEnv.Status.Conditions).To(HaveLen(1))
+			condition := managedEnv.Status.Conditions[0]
+			Expect(condition.Status).To(Equal(metav1.ConditionFalse))
+			Expect(condition.Reason).To(Equal("UnableToLocateContext"))
+			Expect(condition.Message).To(ContainSubstring("the kubeconfig did not have a cluster entry that matched the API URL"))
+		})
+
+		It("should have a connection status condition of True when the connection to the target cluster is successful", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("creating the GitOpsDeploymentManagedEnvironment")
+
+			kubeConfigContents, apiServerURL, err := extractKubeConfigValues()
+			Expect(err).To(BeNil())
+
+			managedEnv, secret := buildManagedEnvironment(apiServerURL, kubeConfigContents)
+
+			k8sClient, err := fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			err = k8s.Create(&secret, k8sClient)
+			Expect(err).To(BeNil())
+
+			err = k8s.Create(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+
+			By("ensuring the managed environment has a connection status condition of True")
+			Eventually(managedEnv, "2m", "1s").Should(HaveStatusCondition(managedgitopsv1alpha1.ManagedEnvironmentStatusConnectionInitializationSucceeded))
+			err = k8s.Get(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+			Expect(managedEnv.Status.Conditions).To(HaveLen(1))
+			condition := managedEnv.Status.Conditions[0]
+			Expect(condition.Status).To(Equal(metav1.ConditionTrue))
+			Expect(condition.Reason).To(Equal("Succeeded"))
+			Expect(condition.Message).To(BeEmpty())
+		})
+	})
+})
+
+func HaveStatusCondition(conditionType string) matcher.GomegaMatcher {
```

Comment:
Can it be moved to tests-e2e/fixture/managedenvironment/fixture.go? since we have similar functions for other CRs defined in same location.